### PR TITLE
feat: indexing on segments

### PR DESCRIPTION
**Pull request by usamoi**
This PR temporarily removes vamana.

Closes , #69, #77, #96, #97.

### CODE REVIEW

File: src/algorithms/impls/vamana.rs
Comment (VoVAllen) [inline src/algorithms/impls/vamana.rs:1]: Please keep the vamana files. Comment it out or make it out of the branch.

File: README.md [hunk 1]
```diff
@@ -186,28 +168,35 @@ There is only one exception: indexes cannot be created on columns without specif
...
+| max_growing_segment_size | integer | Maximum size of unindexed vectors. Default value is `20_000`.       |
+| min_sealed_segment_size  | integer | Minimum size of vectors for indexing. Default value is `1_000`.     |
+| max_sealed_segment_size  | integer | Maximum size of vectors for indexing. Default value is `1_000_000`. |
 
-You can choose only one algorithm in above indexing algorithms. Default value is `hnsw`.
+Options for table `optimizing`.
+
+| Key                | Type    | Description                                                                 |
+| ------------------ | ------- | --------------------------------------------------------------------------- |
+| optimizing_threads | integer | Maximum threads for indexing. Default value is the sqrt of number of cores. |
```
Comment (VoVAllen) [inline README.md:189]: Why use "the sqrt of number of cores" ?

**Review by VoVAllen — COMMENTED** [review]
Also is there any benchmark numbers?

File: README.md [hunk 2]
```diff
@@ -112,37 +112,29 @@ You can create an index, using squared Euclidean distance with the following SQL
...
 
 --- Or using bruteforce with PQ.
 
 CREATE INDEX ON items USING vectors (embedding l2_ops)
 WITH (options = $$
-capacity = 2097152
-[vectors]
-memmap = "disk"
-[algorithm.flat]
+[indexing.flat]
```
Comment (VoVAllen) [inline README.md:121]: Seems a breaking change? Please update the test file also

**Review by VoVAllen — COMMENTED** [review]
Please also add some tests for the configurations added (segment options), and also add some configurations for the quantization options

File: src/algorithms/impls/vamana.rs
Comment (usamoi) [reply src/algorithms/impls/vamana.rs:1]: They do not compile now.

Comment (VoVAllen) [reply src/algorithms/impls/vamana.rs:1]: I think the vamana.rs file is deleted in this pr?

Comment (usamoi) [reply src/algorithms/impls/vamana.rs:1]: Yes.

Comment (VoVAllen) [reply src/algorithms/impls/vamana.rs:1]: Yes. What I mean is to keep those files in the repo, you can put it in a isolated folder if needed

File: README.md (hunk 1 above)
Comment (usamoi) [reply README.md:189]: Just pick a small number.

**Review by VoVAllen — APPROVED** [review]
Great work! Thanks for your effort!

**Pull request merged**